### PR TITLE
Fix SDL doesn't return info param for generic error in case of at least one hmi components doesn't respond

### PR DESCRIPTION
**Pull request by kozlov-oleksandr**
issues: 
[990](https://github.com/smartdevicelink/sdl_core/issues/990)
[1012](https://github.com/smartdevicelink/sdl_core/issues/1012)
[1016](https://github.com/smartdevicelink/sdl_core/issues/1016)

### CODE REVIEW

**Comment by okozlovlux** [comment]
@LuxoftAKutsan @VVeremjova @LitvinenkoIra please review

**Comment by okozlovlux** [comment]
Rebuild required

File: src/components/application_manager/include/application_manager/commands/mobile/perform_audio_pass_thru_request.h [hunk 1]
```diff
@@ -133,10 +133,28 @@ class PerformAudioPassThruRequest : public CommandRequestImpl {
...
+   * @brief Checks result code from HMI for splitted RPC
+   * and returns parameter for sending to mobile app in
+   * audioPassThru communication.
+   * @param ui_response contains result_code from UI
+   * @param tts_response contains result_code from TTS
+   * @param out_result contains result for mobile app
+   * @return result code - 1) UI error code has precedence than TTS's
+   * 2) error_code from TTS is turned to WARNINGS
+   */
+  mobile_apis::Result::eType PrepareAudioPassThruResultCodeForResponse(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/mobile/perform_audio_pass_thru_request.h]: @okozlovlux please return std::pair. It vould be much more explicit, that this function has 2 results:
result_code,  bolean result.

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 2]
```diff
@@ -266,6 +266,33 @@ class CommandRequestImpl : public CommandImpl,
    */
   const CommandParametersPermissions& parameters_permissions() const;
 
+  /**
+   * @brief Adds interface to be awaited for by sdl request command
+     @param interface_id interface which SDL expects to response in given time
+  */
+  void StartAwaitForInterface(const HmiInterfaces::InterfaceID& interface_id);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @okozlovlux interfaceid is enum. Pasing it by reference redundant

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 3]
```diff
@@ -266,6 +266,33 @@ class CommandRequestImpl : public CommandImpl,
...
+
+  /**
+   * @brief Gets interface await state.
+     @param interface_id interface which SDL awaits for response in given time
+     @return true if SDL awaits for response from given interface in
+   interface_id
+  */
+  bool GetInterfaceAwaitState(const HmiInterfaces::InterfaceID& interface_id);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @okozlovlux Get of state cant be boolean value . Please either use enum or rephrase method name

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 4]
```diff
@@ -266,6 +266,33 @@ class CommandRequestImpl : public CommandImpl,
    */
   const CommandParametersPermissions& parameters_permissions() const;
 
+  /**
+   * @brief Adds interface to be awaited for by sdl request command
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @okozlovlux 
"Start waiting for response from 'inderface_id' interface"

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h [hunk 5]
```diff
@@ -266,6 +266,33 @@ class CommandRequestImpl : public CommandImpl,
...
+     @param interface_id interface which SDL expects to response in given time
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @okozlovlux I cant see giving time in method signature. I suppose that this methods does not know anything about time. And they should not. Please remove such from description

File: src/components/application_manager/src/commands/mobile/add_command_request.cc [hunk 6]
```diff
@@ -187,10 +187,12 @@ void AddCommandRequest::Run() {
   }
 
   if (send_ui_) {
+    StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/mobile/add_command_request.cc]: @okozlovlux can it be moved to SendHMIRequest? It is better to make mapping of interfaces andmethod of this interfaces, that inserting additional string to EACH command

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 7]
```diff
@@ -775,6 +776,88 @@ const CommandParametersPermissions& CommandRequestImpl::parameters_permissions()
...
+void CommandRequestImpl::StartAwaitForInterface(
+    const HmiInterfaces::InterfaceID& interface_id) {
+  awaiting_response_interfaces_.insert(interface_id);
+}
+
+bool CommandRequestImpl::GetInterfaceAwaitState(
+    const HmiInterfaces::InterfaceID& interface_id) {
+  std::set<HmiInterfaces::InterfaceID>::const_iterator it =
+      awaiting_response_interfaces_.find(interface_id);
+  return (it != awaiting_response_interfaces_.end());
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @okozlovlux 
`return it != awaiting_response_interfaces_.end();`

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 8]
```diff
@@ -775,6 +776,88 @@ const CommandParametersPermissions& CommandRequestImpl::parameters_permissions()
...
+             : sum + ", " + GetComponentNameFromInterface(container_value);
+}
+
+void CommandRequestImpl::AddTimeOutComponentInfoToMessage(
+    smart_objects::SmartObject& response) const {
+  using NsSmartDeviceLink::NsSmartObjects::SmartObject;
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  if (awaiting_response_interfaces_.empty()) {
+    LOG4CXX_ERROR(logger_, "No interfaces awaiting, info param is empty");
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @okozlovlux IIs it error or normal SDL path?

File: src/components/application_manager/include/application_manager/commands/mobile/perform_audio_pass_thru_request.h (hunk 1 above)
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/commands/mobile/perform_audio_pass_thru_request.h]: @LuxoftAKutsan , fixed cb32748

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 4 above)
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan , fixed cb32748

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 5 above)
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan , fixed cb32748

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 2 above)
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan , fixed cb32748

File: src/components/application_manager/include/application_manager/commands/command_request_impl.h (hunk 3 above)
Comment (okozlovlux) [inline src/components/application_manager/include/application_manager/commands/command_request_impl.h]: @LuxoftAKutsan , fixed cb32748

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 7 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan , fixed 	14c18e6

File: src/components/application_manager/src/commands/command_request_impl.cc (hunk 8 above)
Comment (okozlovlux) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @LuxoftAKutsan , fixed 	14c18e6

**Comment by jacobkeeler** [comment]
Closing as duplicate of #1640

**Pull request closed without merging**